src/app/features/course/course.component.ts

```diff
@@ -17,5 +21,17 @@ export class CourseComponent implements OnInit {
 
   constructor() {}
 
-  ngOnInit(): void {}
+  ngOnInit(): void {
+    this.savedCorces = this.courses;
+  }
+
+  ngDoCheck() {
+    if (this.searchStr) {
+      this.courses = this.courses.filter(
+        (course) => course.title.toLowerCase() === this.searchStr.toLowerCase()
+      );
+    } else if (this.searchStr === '') {
+      this.courses = this.savedCorces;
+    }
```

Comment:
usually we uses pipes in such cases. Also, usually not check so strong. 

`return courses.filter(
c => course.title.toLowerCase().includes(this.searchStr.toLowerCase()
).`

And no need for savedCourses.